File: ImageParsing.js
const request = require('request');
const cheerio = require('cheerio');
const download = require('download-file');
const fs = require('fs');


// 이미지파일 저장 디렉토리
const path = __dirname + "/public/images";
/**
 *  일간 차트의 앨범 사진을 다운하는 함수 getAlbumArt()
 */
function getAlbumArt() {
    let url = 'https://music.bugs.co.kr/chart/track/day/total';

    request(url, (err, response, html) => {
        if(err) { throw err; }

        let $ = cheerio.load(html);

        for(let i=0; i<100; i++) {
            let fileName = i + 1;
            let url = $('a.thumbnail > img').eq(i).attr('src');

            let options = {
                directory: path,
                filename: fileName + '.png'
            };

            download(url, options, (err) => {
                if(err) { throw err; }
            });
        }
    });
}

function deleteAllImage() {
    for(let i=0; i<100; i++) {
        let file = path + '/' + (i+1) + '.png';
        fs.unlink(file, (err) => {
            if(err) { throw err; }
        });
    }
}

File: ImageParsing.py
import os
import shutil
import requests
from bs4 import BeautifulSoup
from urllib import request


URL = 'https://music.bugs.co.kr/chart/track/day/total'
PATH = os.getcwd() + '/static/images/'


# Scrapping html code
def get_html(target_url):
    _html = ""
    response = requests.get(target_url)
    if response.status_code == 200:
        _html = response.text
    return _html


# parse image url and save in list
def get_image_url():
    html = get_html(URL)
    soup = BeautifulSoup(html, 'html.parser')
    img_url = []

    for image in soup.select('a.thumbnail > img'):
        if image.has_attr('src'):
            img_url.append(image.get('src'))
        else:
            continue
    return img_url


# download album art in static/images directory
def download_album_arts():
    images = get_image_url()
    for i in range(0, 100):
        url = images[i]
        file_name = PATH + str(i + 1) + '.png'
        request.urlretrieve(url, file_name)


# delete all album art
def delete_album_art():
    path = os.getcwd() + '/static/images'
    if os.path.exists(path):
        shutil.rmtree(path)
        os.mkdir(path)
    else:
        os.mkdir(path)

